src/services/swap_service.ts

```diff
@@ -508,19 +510,29 @@ export class SwapService {
                     sellToken,
                     apiSwapQuote.buyAmount,
                     apiSwapQuote.sellAmount,
-                    slippagePercentage ?? 0,
                     apiSwapQuote.sources,
+                    slippagePercentage ?? DEFAULT_SLIPPAGE_PERCENTAGE,
                 );
             } else {
-                apiSwapQuote.expectedSlippage = new BigNumber(0);
+                apiSwapQuote.expectedSlippage = null;
             }
 
             if (marketSide === MarketOperation.Sell) {
-                apiSwapQuote.expectedBuyAmount = apiSwapQuote.buyAmount.times(apiSwapQuote.expectedSlippage.plus(1));
+                if (apiSwapQuote.expectedSlippage === null) {
+                    apiSwapQuote.expectedBuyAmount = null;
```

Comment:
TBH I think we want to discuss this response field. I'm not entire sure Matcha is going to use it, and if they are, they could do this calculation themselves.

If we decide to keep this I think let's break it out to a function as it's gotten long in the tooth.

Reply:
I am fine to remove these fields. Matcha probably will use new fields introduced by SlippageProtect project. If they are not, I agree a single `expectedSlippage` is enough for them.